js/main.js

```diff
@@ -254,14 +261,120 @@ function initLegend() {
     legend.addTo(map);
 }
 
+/*
+ * Returns the city name when present in the hash of the current URI;
+ * otherwise the default city name;
+ */
+function getCityName() {
+    var parser = document.createElement('a');
```

Comment:
Is there a reason why you create a separate DOM node for this task? Couldn't you use use [`window.location.hash`](https://stackoverflow.com/a/11662717/857390)?